runtime/compiler/optimizer/InlinerTempForJ9.cpp

```diff
@@ -5100,7 +5100,7 @@ TR_PrexArgInfo* TR_PrexArgInfo::argInfoFromCaller(TR::Node* callNode, TR_PrexArg
    int32_t numArgs = callNode->getNumArguments();
    int32_t numChildren = callNode->getNumChildren();
 
-   TR_PrexArgInfo* argInfo = new (compilation->trStackMemory()) TR_PrexArgInfo(numArgs, compilation->trMemory());
+   TR_PrexArgInfo* argInfo = new (compilation->trHeapMemory()) TR_PrexArgInfo(numArgs, compilation->trMemory());
```

Comment:
Just curious why the scope of the allocation has changed.  Does `argInfo` now have a longer lifetime than it used to, or does it "escape" ?

Reply:
This util is only used in actually inlining, with this PR, it'll be used in ECS (estimate code size) as well. ECS runs in stack memory region but arg info created in ECS is stored to `_prexArgInfo` in calltargets and callsites, which is allocated in heap memory and will be used after ECS. Although `_prexArgInfo` in calltargets or callsites are not used later than ECS now, allocating it in stack memory region will make them stale after ECS. Wrong decision might be made if someone wants to access them after ECS.
https://github.com/eclipse/openj9/blob/15fced5022e0bb562f122a195efcca915795c234/runtime/compiler/optimizer/EstimateCodeSize.cpp#L91

Notice that the array of TR_PrexArgument is always allocated on heap, maybe we should have arg info, prex argument, call target and call site all allocated in a region that will be destroyed when inliner finishes?
https://github.com/eclipse/omr/blob/1fdf296130d4313dd405a50cd770c4da181164ca/compiler/optimizer/PreExistence.hpp#L135

Reply:
I don't know if this additional complexity wrt memory and lifetime management will be worth it since I don't believe that these data structures contribute to our JIT compile time footprint in a significant manner.